flowforge/input/BoundaryConditions.py

```diff
@@ -112,17 +116,140 @@ def type_outlet(self):
 
     def _convertUnits(self, uc: UnitConverter) -> None:
         if self._val_inlet:
-            if self.type_inlet == 'temperature':
+            if self.type_inlet == "temperature":
                 self._val_inlet = uc.temperatureConversion(self._val_inlet)
-            elif self.type_inlet == 'enthalpy':
+            elif self.type_inlet == "enthalpy":
                 self._val_inlet *= uc.enthalpyConversion
             else:
                 raise Exception("Unknown enthalpy BC type: " + self.type_inlet)
 
         if self._val_outlet:
-            if self.type_outlet == 'temperature':
+            if self.type_outlet == "temperature":
                 self._val_outlet = uc.temperatureConversion(self._val_outlet)
-            elif self.type_outlet == 'enthalpy':
+            elif self.type_outlet == "enthalpy":
                 self._val_outlet *= uc.enthalpyConversion
             else:
                 raise Exception("Unknown enthalpy BC type: " + self.type_outlet)
+
+
+class BoundaryConditions:
+    """
+    Continer class for all input boundary conditions
+
+    "bounday_conditions" should be defined as a dict in the form:
+
+    bounday_conditions = {
+        "unique_boundary_name" :
+                {"boundary_type": "DirichletBC", "surface": "surface_name", "varaible": "varaible_name",  "value", float},
+        "inlet_mdot"           :
+                {"boundary_type": "DirichletBC", "surface": "inlet",        "varaible": "mass_flow_rate", "value", 25.0},
+        "outlet_pressure"      :
+                {"boundary_type": "DirichletBC", "surface": "outlet",       "varaible": "pressure",       "value", 1e5},
+        "inlet_temperature"    :
+                {"boundary_type": "DirichletBC", "surface": "inlet",        "varaible": "temperature",    "value", 700}
+    }
+    """
+    def __init__(self, **bounday_conditions: dict):
+        bc_names = list(bounday_conditions.keys())
+
+        bc_objects = {"DirichletBC": DirichletBC}
+
+        bc_list = []
+        for bc_name in bc_names:
+            bc_type = bounday_conditions[bc_name]["boundary_type"]
+            bc_obj = bc_objects[bc_type]
+            bc = bc_obj(bounday_conditions["surface"],
+                        bounday_conditions["variable"],
+                        bounday_conditions["value"])
+            bc_list.append(bc)
```

Comment:
I think you're missing a `[bc_name]` when retrieving `surface`, `variable`, and `value` off the `boundary_conditions` dictionary.

Actually, I'm wondering why you didn't do something like this:

```
self.bcs = {}
for bc_name, bc in bound_conditions.items():
    bc_type = bc["boundary_type"]
    bc_obj = bc_objects[bc_type]
    self.bcs[bc_name] = bc_obj(bc["surface"], bc["variable"], bc["value"]))
```

Reply:
I have gone back through and replaced it with this. This is a much more compact way to do exactly what I wanted - thanks!

I'll also go through my syth code and make sure I don't have any similar issues, and generally clean that up once this PR is complete.